src/full/Agda/TypeChecking/Monad/Base.hs

```diff
@@ -2313,6 +2322,8 @@ data RewriteRule = RewriteRule
   , rewRHS     :: Term       -- ^ @Γ ⊢ rhs : t@.
   , rewType    :: Type       -- ^ @Γ ⊢ t@.
   , rewFromClause :: Bool    -- ^ Was this rewrite rule created from a clause in the definition of the function?
+  , rewTopModule  :: TopLevelModuleName
+      -- ^ In which file is this rewrite rule defined?
```

Comment:
Maybe add to this command the purpose of this field: to filter rewrite rules from the imported signature.